pkg/k8s/watchers/cilium_node.go

```diff
@@ -65,104 +63,54 @@ type K8sCiliumNodeWatcher struct {
 	ciliumNodeStore atomic.Pointer[resource.Store[*cilium_v2.CiliumNode]]
 }
 
-func (k *K8sCiliumNodeWatcher) ciliumNodeInit(ctx context.Context, asyncControllers *sync.WaitGroup) {
-	// CiliumNode objects are used for node discovery until the key-value
-	// store is connected
-	var once sync.Once
-	apiGroup := k8sAPIGroupCiliumNodeV2
-
-	for {
-		var synced atomic.Bool
-		stop := make(chan struct{})
-
-		k.k8sResourceSynced.BlockWaitGroupToSyncResources(
-			stop,
-			nil,
-			func() bool { return synced.Load() },
-			apiGroup,
-		)
-		k.k8sAPIGroups.AddAPI(apiGroup)
-
-		// Signalize that we have put node controller in the wait group to sync resources.
-		once.Do(asyncControllers.Done)
-
-		// derive another context to signal Events() and Store() in case of kvstore connection
-		subCtx, cancel := context.WithCancel(ctx)
-
-		var wg sync.WaitGroup
-
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			defer close(stop)
-
-			events := k.resources.CiliumNode.Events(subCtx)
-			cache := make(map[resource.Key]*cilium_v2.CiliumNode)
-			for event := range events {
-				var err error
-				switch event.Kind {
-				case resource.Sync:
-					synced.Store(true)
-					k.nodeManager.NodeSync()
-				case resource.Upsert:
-					var needUpdate bool
-					oldObj, ok := cache[event.Key]
-					if !ok {
-						needUpdate = k.onCiliumNodeInsert(event.Object)
-					} else {
-						needUpdate = k.onCiliumNodeUpdate(oldObj, event.Object)
+func (k *K8sCiliumNodeWatcher) ciliumNodeInit(ctx context.Context) {
+	var synced atomic.Bool
+
+	k.k8sResourceSynced.BlockWaitGroupToSyncResources(
+		ctx.Done(),
+		nil,
```

Comment:
why do we need this if we don't pass a wait group?

Reply:
Yes, the mention to the wait group is a bit misleading, but that function registers the synced func and eventually marks the given resource as synchronized. That is used in the daemon initialization to block until all appropriate informers are synchronized.

https://github.com/cilium/cilium/blob/1914c208c0a9dedb33e6e046265eabd87d5e532c/daemon/cmd/daemon_main.go#L1675-L1683